analyzers/src/SonarAnalyzer.CSharp/Rules/VariableShadowsField.cs

```diff
@@ -90,7 +90,8 @@ private static List<ISymbol> GetContextSymbols(SonarSyntaxNodeReportingContext c
 
         private static void ReportOnVariableMatchingField(SonarSyntaxNodeReportingContext context, IEnumerable<ISymbol> members, SyntaxToken identifier)
         {
-            if (members.FirstOrDefault(x => x.Name == identifier.ValueText) is { } matchingMember)
+            if (members.FirstOrDefault(x => x.Name == identifier.ValueText
+                && (x.IsStatic || !identifier.Parent.EnclosingScope().GetModifiers().Any(x => x.Kind() == SyntaxKind.StaticKeyword))) is { } matchingMember)
```

Comment:
Despite [this comment on the issue](https://github.com/SonarSource/sonar-dotnet/issues/8260#issuecomment-1781344472), I did not use the [SemanticModel.IsAccesible()](https://learn.microsoft.com/de-de/dotnet/api/microsoft.codeanalysis.semanticmodel.isaccessible?view=roslyn-dotnet-4.6.0) method.
Reason:
1. The rule only checks for overshadowed members in the same type declaration, it doesn't check for members declared in base types.
2. [SemanticModel.IsAccesible() only checks for access modifiers](accessibility modifiers), it doesn't consider static context.